### PR TITLE
isns: add open-isns

### PR DESCRIPTION
**Pull request by lucize**
prerequisite for open-iscsi

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

Maintainer: me 
Tested x86

needed for open-iscsi

### CODE REVIEW

**Comment by neheb** [comment]
Just tested this. It does not compile without deprecated OpenSSL APIs.

PKG_SOURCE is totally bogus. I'd rather that codeload be used.

**Comment by neheb** [comment]
```
--- a/pki.c
+++ b/pki.c
@@ -13,6 +13,8 @@
 #include <openssl/pem.h>
 #include <openssl/err.h>
 #include <openssl/evp.h>
+#include <openssl/dsa.h>
+#include <openssl/bn.h>
 #endif
 #include <fcntl.h>
 #include <libisns/isns.h>
@@ -96,13 +98,11 @@ isns_create_dsa_context(void)
 	isns_security_t	*ctx;
 
 	if (!isns_openssl_init) {
-		ERR_load_crypto_strings();
 #if OPENSSL_API_COMPAT < 0x10100000L
+		ERR_load_crypto_strings();
 		OpenSSL_add_all_algorithms();
 		OpenSSL_add_all_ciphers();
 		OpenSSL_add_all_digests();
-#else
-		OPENSSL_init_crypto();
 #endif
 		isns_openssl_init = 1;
 	}
```

Rough patch

File: net/open-isns/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+PKG_LICENSE_FILES:=COPYING
+
+PKG_FIXUP:=autoreconf
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/open-isns
+	SECTION:=libs
```
Comment (neheb) [inline net/open-isns/Makefile:24]: replace TAB with two spaces for all.

File: net/open-isns/Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	TITLE:=open-isns
+	DEPENDS:=+libopenssl
+endef
+
+define Package/open-isns/description
+This is a partial implementation of iSNS, according to RFC4171.
+endef
+
+CONFIGURE_ARGS += \
+    --prefix=/usr \
```
Comment (neheb) [inline net/open-isns/Makefile:36]: replace spaces with TAB.

Only --enable-shared is needed.

**Comment by neheb** [comment]
What does -Werror error on?

**Comment by lucize** [comment]
actually seems that is not doing it, should I let -Werror and no apply the patch or let the patch ?

File: net/open-isns/Makefile (hunk 2 above)
Comment (diizzyy) [reply net/open-isns/Makefile:37]: Not needed
https://github.com/openwrt/openwrt/blob/master/include/package-defaults.mk#L80
LIB_DIR isn't defined so I have no opinion about that one.

**Comment by neheb** [comment]
Yeah get rid of the configure.ac patch. Also remove autoreconf.

I removed both locally and it compiled.

**Comment by neheb** [comment]
Oh wait this is an upstream backport. Never mind.

**Pull request merged**